src/services/usersService.js

```diff
@@ -6,13 +6,61 @@ import { history } from "@helpers/history";
 import validateEmail from "@validations/user/validateEmail";
 import validateFieldLength from "@validations/validateFieldLength";
 import validateIsEmptyObject from "@validations/validateIsEmtpyObject";
+import validateLogin from "@validations/user/validateLogin";
+import validatePassword from "@validations/user/validatePassword";
 import validatePhone from "@validations/user/validatePhone";
 
 class UsersService {
   constructor() {
     this.currentUser = store.getState().currentUser.value;
   }
 
+  async login(data) {
+    const { login, password } = data;
+
+    if (!login?.length || !password?.length) {
+      return "The login and password fields must not be empty.";
+    }
+
+    if (!validateLogin(login)) {
+      return "The login field must contain from 3 to 20 characters.";
+    }
+
+    if (!validatePassword(password)) {
+      return "The password field must contain from 3 to 20 characters";
+    }
+
+    const { token: userToken, user: userData } = await api.userLogin({
+      login: login.trim().toLowerCase(),
+      password: password.trim(),
+    });
+    localStorage.setItem("sessionId", userToken);
```

Comment:
just curious, why we use local storage for sessionId, but use var for curerntUserId ?
any particular reason for not using something similar?

Reply:
The user id looks less secure than a token, so storing it locally in the service seemed more appropriate to me, and keeping it in a variable allows us to avoid regular decoding of the token if the user id is needed in methods, which was the main factor for choosing it.